p2p/src/node.rs

```diff
@@ -640,14 +645,16 @@ where
                             self.swarm.add_external_address(info.observed_addr.clone());
                         } else if let Some(autonat) = self.swarm.behaviour_mut().autonat.as_mut() {
                             // Probe the observed addr for external connectivity.
-                            // See OutboundProbeEvent case for
-
-                            debug!(
-                                address=%info.observed_addr,
-                                %peer_id,
-                                "probing observed address from peer for external connectivity",
-                            );
-                            autonat.probe_address(info.observed_addr.clone());
+                            // Only probe one address at a time.
+                            if self.active_address_probe.is_none() {
```

Comment:
if there's already a probe running, then it looks like we drop this addr on the floor.  Should we insert it into a queue to probe later?

Reply:
I thought about it but I don't think its needed. Everytime we connect to a peer it will report the observed address and trigger this code. So this logic is run very frequently. Additionally its typical to have only a single external address. In which case its likely that the in-progress address probe is the correct one. In cases where there are lots of different observed addresses its likely that NAT hasn't been setup correctly for the peer and so doesn't have an external address.  

Maybe I could add this as a comment on in the code?